src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,181 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        validateIndex(index,false);
+        newLink(value,index);
```

Comment:
Create linkBefore method
```suggestion
        if (index == size) {
            add(value);
        } else {
            Node<T> target = getNode(index);
            linkBefore(value, target);
        }
```